Sources/NodeZooSDK/Internal/APIService.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  APIService.swift
+//  
+//
+//  Created by Pankaj Kulkarni on 09/05/23.
+//
+
+import Foundation
+
+protocol APIServicingClosure {
+    func search(request: URLRequest, completion: @escaping (Result<[NZPackage], Error>)->Void) -> URLSessionDataTask?
+}
+
+struct ClosureBasedAPIService: APIServicingClosure {
+    
+    var urlSession = URLSession.shared
+
+    
+    func search(request: URLRequest, completion: @escaping (Result<[NZPackage], Error>) -> Void) -> URLSessionDataTask? {
+        
+        let dataTask = urlSession.dataTask(with: request) {data, response, error in
+            print("Received response")
+            guard let httpResponse = response as? HTTPURLResponse else {
+                completion(.failure(NodeZooServiceError.invalidResponse))
+                return
+            }
+            
+            // --- Debugging purpose only ---
```

Comment:
Please add an option flag to turn this on and off

Reply:
Done